pkg/disk/disk.go

```diff
@@ -0,0 +1,138 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package disk manages the persistent disk used to save containerd user data
+package disk
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+
+	"github.com/spf13/afero"
+
+	"github.com/runfinch/finch/pkg/command"
+	"github.com/runfinch/finch/pkg/path"
+)
+
+const (
+	// diskName must always be consistent with the value under additionalDisks in finch.yaml.
+	diskName = "finch"
+)
+
+// UserDataDiskManager is used to check the user data disk configuration and create it if needed.
+type UserDataDiskManager interface {
+	InitializeUserDataDisk() error
+}
+
+// fs functions required for setting up the user data disk.
+type diskFS interface {
+	afero.Fs
+	afero.Linker
+	afero.LinkReader
+}
+
+type userDataDiskManager struct {
+	lcc     command.LimaCmdCreator
+	fs      diskFS
+	finch   path.Finch
+	homeDir string
+}
+
+// NewUserDataDiskManager is a constructor for UserDataDiskManager.
+func NewUserDataDiskManager(
+	lcc command.LimaCmdCreator,
+	fs diskFS,
+	finch path.Finch,
+	homeDir string,
+) UserDataDiskManager {
+	return &userDataDiskManager{
+		lcc:     lcc,
+		fs:      fs,
+		finch:   finch,
+		homeDir: homeDir,
+	}
+}
+
+// InitializeUserDataDisk checks the current disk configuration and fixes it if needed.
+func (m *userDataDiskManager) InitializeUserDataDisk() error {
+	if m.limaDiskExists() {
+		limaPath := fmt.Sprintf("%s/_disks/%s/datadisk", m.finch.LimaHomePath(), diskName)
+		loc, err := m.fs.ReadlinkIfPossible(limaPath)
+		if err != nil {
+			return err
+		}
+		// if the file is not m symlink, loc will be an empty string
```

Comment:
nit: typo? Should this read: if the file is not ~~m~~ **a** symlink? I might just be understanding it wrong

Reply:
yeah, should be **a** symlink. I had previously used `a` as a temporary variable name and I guess my IDE got a little overzealous when I renamed the variable 😆